File: day-1/part_2.rb
INPUT_FILE = "input_1.txt"

def solve()
    elves = Array.new
    current_calories = 0
    f = File.open(INPUT_FILE, "r") do |f|
        f.each { |line|
            line.chomp!
            if line == ""
                elves << current_calories 
                current_calories = 0
            else
                current_calories += Integer(line)
            end
        }
    end
    elves.sort!
    elves.reverse!
    puts elves[0] + elves[1] + elves[2]
end

solve

File: day-1/part_2.py
INPUT_FILE = "input_1.txt"


def solve():
    elves = []
    current_calories = 0
    with open(INPUT_FILE, "r") as input_file:
        for row in input_file:
            row = row.strip("\n")
            if row == "":
                elves.append(current_calories)
                current_calories = 0
            else:
                current_calories += int(row)
    elves.sort(reverse=True)
    print(elves[0] + elves[1] + elves[2])


if __name__ == "__main__":
    solve()
